classes/sso/hash/hashing_service.php

```diff
@@ -50,6 +50,20 @@ public static function generate_hash(string $user, \stdClass $mumietask) : mumie
         return $mumieidhash;
     }
 
+    /**
+     * Generate a hash of the userid with a lecturer postfix.
+     *
+     * @param string $user The user id for which to generate the hash
+     * @return mumie_id_hash The generated mumie_id_hash
+     */
+    public static function generate_hash_with_lecturer_postfix(string $user) : mumie_id_hash {
+        $hash = auth_mumie_get_hashed_id($user);
+        $hash .= '@lecturer@';
+        $mumieidhash = new mumie_id_hash($user, $hash);
+        $mumieidhash->save();
```

Comment:
Do we need to save here?

Reply:
We need to save, because this database is queried for this information when the verify token endpoint (`verifyToken.php`) is called. It is used to get the Moodle user id for a MUMIE user id, since you can't decipher the hash.